internal/provider/cluster_resource.go

```diff
@@ -377,6 +376,27 @@ func (r *clusterResource) ConfigValidators(_ context.Context) []resource.ConfigV
 	}
 }
 
+func derivePlanType(cluster *CockroachCluster) (client.PlanType, error) {
+	var planType client.PlanType
+	if IsKnown(cluster.Plan) {
+		planType = client.PlanType(cluster.Plan.ValueString())
+	} else if cluster.DedicatedConfig != nil {
+		planType = client.PLANTYPE_ADVANCED
+	} else if cluster.ServerlessConfig != nil {
+		if cluster.ServerlessConfig.UsageLimits != nil && IsKnown(cluster.ServerlessConfig.UsageLimits.ProvisionedVirtualCpus) {
```

Comment:
Is `IsKnown` the same as non-zero? Do we reject specs that set a usage limit and also explicitly set `provisioned_virtual_cpus == 0`?

Reply:
Oops I meant to delete this. I think the `ConfigValidators` on the cluster resource guarantee that they can't be set together.